### PR TITLE
Add proj6

### PR DESCRIPTION
**Pull request by you06**
Add project 6 which focus on the distributed transaction.

### CODE REVIEW

**Comment by you06** [comment]
@rebelice can you help review when you have time?

File: courses/proj6-README-zh_CN.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## 执行模型介绍
+
+在 TiDB 中，事务的执行过程会被缓存在 buffer 中，在提交时，才会通过 Percolator 提交协议将其完整的写入到分布的 TiKV 存储引擎中。这一调用的入口是 `store/tikv/txn.go` 中的 `tikvTxn.Commit` 函数。
+
+在执行时，一个事务可能会遇到其他执行过程中的事务，此时需要通过 Lock Resolve 组件来查询所遇到的事务状态，并根据查询到的结果执行相应的措施。
+
+## Two Phase Commit
+
+Percolator 提交协议的两阶段提交分为 Prewrite 和 Commit，其中 Prewrite 实际写入数据，Commit 让数据对外可见。其中，事务的成功以 Primary Key 为原子性标记，当 Prewrite 失败或是 Primary Key Commit 失败时需要进行垃圾清理，将写入的事务回滚。
```
Comment (rebelice) [inline courses/proj6-README-zh_CN.md]:
```suggestion
Percolator 提交协议的两阶段提交分为 Prewrite 和 Commit：Prewrite 实际写入数据，Commit 让数据对外可见。其中，事务的成功以 Primary Key 为原子性标记。当 Prewrite 失败或是 Primary Key Commit 失败时需要进行垃圾清理，将写入的事务回滚。
```

File: courses/proj6-README-zh_CN.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+在执行过程中，会涉及到三类操作，分别是 Prewrite/Commit/Rollback(Cleanup)。这些操作会在同一个流程中被处理。你需要完成 Prewrite 过程中的 `buildPrewriteRequest` 函数，然后仿照 Prewrite 的 `handleSingleBatch` 函数完成 Commit 和 Rollback 的 `handleSingleBatch` 函数。
+
+## Lock Resolver
+
+在 Prewrite 阶段，对于一个 Key 的操作会写入两条记录。
+
+- Default CF 中存储了实际的 KV 数据。
+- Lock CF 中存储了锁，包括 Key 和时间戳信息，会在 Commit 成功时清理。
+
+Lock Resolver 的职责就是当一个事务在提交过程中遇到 Lock 时，需要如何应对。
```
Comment (rebelice) [inline courses/proj6-README-zh_CN.md]:
```suggestion
Lock Resolver 的职责就是应对一个事务在提交过程中遇到 Lock 的情况。
```

**Review by rebelice — COMMENTED** [review]
How about adding some information about the Percolator protocol as a reference? For example, the paper introducing Percolator. 
Rest LGTM.

**Comment by you06** [comment]
@rebelice I added some description about the Percolator protocol.

**Review by rebelice — APPROVED** [review]

**Pull request merged**